targets/unity-v2/source/Authentication/Strategies/SilentAuthStrategy.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using PlayFab.ClientModels;
+using UnityEngine;
+
+namespace PlayFab.Authentication.Strategies
+{
+    internal class SilentAuthStrategy : IAuthenticationStrategy
+    {
+        public virtual void Authenticate(PlayFabAuthService authService, Action<LoginResult> resultCallback, Action<PlayFabError> errorCallback)
+        {
+#if UNITY_ANDROID && !UNITY_EDITOR
+            PlayFabClientAPI.LoginWithAndroidDeviceID(new LoginWithAndroidDeviceIDRequest
+            {
+                TitleId = PlayFabSettings.TitleId,
+                AndroidDevice = SystemInfo.deviceModel,
+                OS = SystemInfo.operatingSystem,
+                AndroidDeviceId = PlayFabSettings.DeviceUniqueIdentifier,
+                CreateAccount = true,
+                InfoRequestParameters = authService.InfoRequestParams
+            }, resultCallback, errorCallback);
+#elif UNITY_IPHONE || UNITY_IOS && !UNITY_EDITOR
+            PlayFabClientAPI.LoginWithIOSDeviceID(new LoginWithIOSDeviceIDRequest
+            {
+                TitleId = PlayFabSettings.TitleId,
+                DeviceModel = SystemInfo.deviceModel,
+                OS = SystemInfo.operatingSystem,
+                DeviceId = SystemInfo.deviceUniqueIdentifier,
+                CreateAccount = true,
+                InfoRequestParameters = authService.InfoRequestParams
+            }, resultCallback, errorCallback);
+#else
+            PlayFabClientAPI.LoginWithCustomID(new LoginWithCustomIDRequest
+            {
+                TitleId = PlayFabSettings.TitleId,
+                CustomId = SystemInfo.deviceUniqueIdentifier,
+                CreateAccount = true,
+                InfoRequestParameters = authService.InfoRequestParams
+            }, resultCallback, errorCallback);
+#endif
+        }
+
+        public virtual void Link(PlayFabAuthService authService)
```

Comment:
You should note that your deviceIds mismatched between Authenticate, and Link.
Be very careful with that.